meterfs/meterfs.c

```diff
@@ -634,10 +657,20 @@ int meterfs_allocateFile(const char *name, size_t sectorcnt, size_t filesz, size
 		meterfs_powerctrl(0, ctx);
 		return err;
 	}
-	meterfs_powerctrl(0, ctx);
+
 	ctx->filecnt += 1;
 	ctx->hcurrAddr = headerNew;
 
+	/* Add new node to data structure. */
+	meterfs_getFilePos(&f, ctx);
+
+	if ((err = node_add(&f, (id_t)ctx->filecnt, &ctx->nodesTree)) < 0) {
+		meterfs_powerctrl(0, ctx);
+		return err;
+	}
+
+	meterfs_powerctrl(0, ctx);
```

Comment:
No need for power to be on during node add. You can turn the power off earlier (just after `meterfs_getFilePos`)